website/identifiers/metadata.py

```diff
@@ -1,127 +0,0 @@
-# -*- coding: utf-8 -*-
-import unicodedata
-import lxml.etree
-import lxml.builder
-
-from website import settings
-
-NAMESPACE = 'http://datacite.org/schema/kernel-4'
-XSI = 'http://www.w3.org/2001/XMLSchema-instance'
-SCHEMA_LOCATION = 'http://datacite.org/schema/kernel-4 http://schema.datacite.org/meta/kernel-4/metadata.xsd'
-E = lxml.builder.ElementMaker(nsmap={
-    None: NAMESPACE,
-    'xsi': XSI},
-)
-
-CREATOR = E.creator
-CREATOR_NAME = E.creatorName
-SUBJECT_SCHEME = 'bepress Digital Commons Three-Tiered Taxonomy'
-
-# From https://stackoverflow.com/a/19016117
-# lxml does not accept strings with control characters
-def remove_control_characters(s):
-    return ''.join(ch for ch in s if unicodedata.category(ch)[0] != 'C')
-
-# This function is not OSF-specific
-def datacite_metadata(doi, title, creators, publisher, publication_year, pretty_print=False):
-    """Return the formatted datacite metadata XML as a string.
-
-    :param str doi
-    :param str title
-    :param list creators: List of creator names, formatted like 'Shakespeare, William'
-    :param str publisher: Publisher name.
-    :param int publication_year
-    :param bool pretty_print
-    """
-    creators = [CREATOR(CREATOR_NAME(each)) for each in creators]
-    root = E.resource(
-        E.resourceType('Project', resourceTypeGeneral='Text'),
-        E.identifier(doi, identifierType='DOI'),
-        E.creators(*creators),
-        E.titles(E.title(remove_control_characters(title))),
-        E.publisher(publisher),
-        E.publicationYear(str(publication_year)),
-    )
-    # set xsi:schemaLocation
-    root.attrib['{%s}schemaLocation' % XSI] = SCHEMA_LOCATION
-    return lxml.etree.tostring(root, pretty_print=pretty_print)
```

Comment:
Can you tell me what we used to use this for, and what you're using now instead? Same with `datacite_metadata_for_node`?

Reply:
This was originally conceived as part of SHARE I believe. OSF and SHARE were going to share these xml parsing tools, but it never materialized.